sale_promotion_programs/models/sale_coupon_program.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, api, fields, models
+
+
+class SaleCouponProgram(models.Model):
+    _inherit = "sale.coupon.program"
+
+    # Add posibility to use discount only on first customer"s sale order
+    first_order_only = fields.Boolean(
+        string="Apply only on the first order of each client matching the conditions",
+        default=True,
+    )
+
+    def _check_promo_code(self, order, coupon_code):
```

Comment:
you should not reimplement completely the method -> you need to call super() for the checks which are part of the sale_coupon module and only add checks specific to this module in the overriden method.